tests/test_callbacks.cpp

```diff
@@ -240,4 +240,38 @@ TEST_SUBMODULE(callbacks, m) {
             f();
         }
     });
+
+    auto *custom_def = []() {
+        static PyMethodDef def;
+        def.ml_name = "example_name";
+        def.ml_doc = "Example doc";
+        def.ml_meth = [](PyObject *, PyObject *args) -> PyObject * {
+            if (PyTuple_Size(args) != 1) {
+                throw std::runtime_error("Invalid number of arguments for example_name");
+            }
+            PyObject *first = PyTuple_GetItem(args, 0);
+            if (!PyLong_Check(first)) {
+                throw std::runtime_error("Invalid argument to example_name");
+            }
+            auto result = py::cast(PyLong_AsLong(first) * 9);
+            return result.release().ptr();
+        };
+        def.ml_flags = METH_VARARGS;
+        return &def;
+    }();
+
+    // rec_capsule with custom name
```

Comment:
Question before maybe making a suggestion, is the point here:
* the capsule name matches that defined in `internals`,
* but it's a different pointer?

Reply:
Yep. This is how we ensure ABI safety. Only exact pointer matches are allowed.

Reply:
If we were using unsafe string matching (and not the safe pointer matching), this test would fail and cause a default.

Reply:
Cool, thanks!

I'm worried about the names here and in internals.h getting out of sync in the long run.

Suggestion:

In internals.h, `namespace pybind11::detail`:
```
static constexpr const char *internals_function_record_capsule_name = "pybind11_function_record_capsule";
```

Then use that in internals.h and here.

Add a comment here, something like:

```
// Ensure that the capsule is not accidentally used as a pybind11 function_record, even if the name matches.
```

Reply:
Done!